File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
public class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        String alphabet = "abcdefghijklmnopqrstuvwxyz";
        int idx;

        for(int i =0; i<skip.length(); i++) {
            alphabet = alphabet.replace(Character.toString(skip.charAt(i)), "");
        }

        for(int i =0; i<s.length(); i++) {
            idx = alphabet.indexOf(Character.toString(s.charAt(i))) + index;
            answer += Character.toString(alphabet.charAt(idx % alphabet.length()));
        }
        return answer;
    }


    public static void main(String[] args) {
        Solution sol = new Solution();
        String s = "aukks";
        String skip = "wbqd";
        int index = 5;
        String answer = sol.solution(s, skip, index);
        System.out.println("answer = " + answer);
    }
}

File: JavaScript/프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    const skipAsciiCode = [...skip].map(char => char.charCodeAt(0));
    const answer = [...s].map(char => {
        let newAscii = char.charCodeAt(0);

        for (let i = 0; i < index; i++) {
            newAscii++;
            if (newAscii > 122) newAscii = 97;
    
            while (skipAsciiCode.includes(newAscii)) {
                newAscii++;
                if (newAscii > 122) newAscii = 97;
            }
        }
        return String.fromCharCode(newAscii);
    }).join('');

    return answer;
}